post_processing/jacobian_plots.jl

```diff
@@ -0,0 +1,357 @@
+import ClimaComms
+import ClimaCore.InputOutput: HDF5Reader
+import LinearAlgebra: diag
+
+function make_jacobian_plots(output_path, Yₜ_end, dt)
+    context = ClimaComms.context(Yₜ_end.c)
+    file_paths = filter(endswith(".hdf5"), readdir(output_path; join = true))
+    for file_path in file_paths
+        file_name = split(basename(file_path), '.')[1]
+        contains(file_name, "jacobian") || continue
+        time_series = read_time_series(file_path, context)
+        plot_jacobian(output_path, time_series, file_name, Yₜ_end, dt)
+        if startswith(file_name, "approx_jacobian")
+            exact_file_path = joinpath(
+                output_path,
+                replace(basename(file_path), "approx" => "exact"),
+            )
+            if exact_file_path in file_paths
+                exact_time_series = read_time_series(exact_file_path, context)
+                error_time_series = map(
+                    time_series,
+                    exact_time_series,
+                ) do approx_data, exact_data
+                    (t, approx_title, approx_∂Yₜ_∂Y, approx_rescaled_∂Yₜ_∂Y, Yₜ) = approx_data
+                    (_, _, exact_∂Yₜ_∂Y, exact_rescaled_∂Yₜ_∂Y, _) =
+                        exact_data
+                    @assert exact_data[1] == t
+                    @assert exact_data[5] == Yₜ
+                    error_title = "Error of $approx_title"
+                    error_∂Yₜ_∂Y = approx_∂Yₜ_∂Y .- exact_∂Yₜ_∂Y
+                    error_rescaled_∂Yₜ_∂Y =
+                        approx_rescaled_∂Yₜ_∂Y .- exact_rescaled_∂Yₜ_∂Y
+                    (t, error_title, error_∂Yₜ_∂Y, error_rescaled_∂Yₜ_∂Y, Yₜ)
+                end
+                error_file_name =
+                    replace(file_name, "jacobian" => "jacobian_error")
+                plot_jacobian(
+                    output_path,
+                    error_time_series,
+                    error_file_name,
+                    Yₜ_end,
+                    dt,
+                )
+            end
+        end
+    end
+end
+
+function read_time_series(file_path, context)
+    reader = HDF5Reader(file_path, context)
+    t_strings =
+        sort(keys(reader.file); by = t_string -> parse(Float64, t_string))
+    time_series = map(t_strings) do t_string
+        data_at_t = reader.file[t_string]
+        title = data_at_t["title"][]
+        ∂Yₜ_∂Y = data_at_t["∂Yₜ_∂Y"][]
+        Yₜ = data_at_t["Yₜ"][]
+
+        safe_inv(x) = iszero(x) ? zero(x) : inv(x)
+        rescaled_∂Yₜ_∂Y = safe_inv.(Yₜ) .* ∂Yₜ_∂Y .* Yₜ'
+
+        # Take the transpose of each matrix so that its rows are plotted along
+        # the y-axis and its columns are plotted along the x-axis.
+        (parse(Float64, t_string), title, ∂Yₜ_∂Y', rescaled_∂Yₜ_∂Y', Yₜ)
+    end
+    Base.close(reader)
```

Comment:
I'd recommend using the `do`-syntax, as that will do a `try-catch-finally`, and close the file in the case that something in here breaks, preventing memory leaks.